mail_deduplicate/mail.py

```diff
@@ -199,6 +199,24 @@ def hash_key(self):
         logger.debug(f"Hash: {hash_value}")
         return hash_value
 
+    @cachedproperty
+    def hash_raw_body(self):
+        """ Returns the canonical body hash of a mail. """
+        serialized_raw_body = "\n".join(self.body_lines).encode("utf-8")
+        hash_value = hashlib.sha224(serialized_raw_body).hexdigest()
+        logger.debug(f"Body raw hash: {hash_value}")
+        return hash_value
+
+    @cachedproperty
+    def hash_normalized_body(self):
+        """ Returns the normalized body hash of a mail. """
+        serialized_normalized_body = "".join(
+            [re.sub(r"\s", "", l) for l in self.body_lines]
```

Comment:
I guess you should call `self.body_lines.split()` here instead of the regexp. `split()`, without parameters, is the simplest (faster?) tokenizer out there:
```python
>>> " sdfsaff dsfsd  sf        sdfsdf \n  dsdsd  \t axdsad ".split()
['sdfsaff', 'dsfsd', 'sf', 'sdfsdf', 'dsdsd', 'axdsad']
```

Reply:
I honestly don't know about the faster but I felt like the `re` solution would be more readable.
Using `split()`, there will even need a second `join()`:
```
serialized_normalized_body = "".join(
    [re.sub(r"\s", "", l) for l in self.body_lines]
).encode("utf-8")
```
would become:
```
serialized_normalized_body = "".join(
    ["".join(l.split()) for l in self.body_lines]
).encode("utf-8")
```

Apart from that, both solution seems valid to me. Give your order and I'll update the pull request 😉

Reply:
I tend to avoid regexps if I can have a better way of expressing the same effect with plain Python. Mostly for readability. But OK, it's good to me. Let's not waste too much time on this. Thanks again @NCenerar for the update! :)